### PR TITLE
Microsoft at machines

### PR DESCRIPTION
**Pull request by orhovy**
Draft for microsoft ATP bug:
Both timeout and paging.
https://github.com/demisto/etc/issues/34362
https://github.com/demisto/etc/issues/34245

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 1]
```diff
@@ -222,15 +222,19 @@ def unisolate_machine(self, machine_id, comment):
...
 
-    def get_machines(self, filter_req):
+    def get_machines(self, filter_req, next_link, page_size):
         """Retrieves a collection of Machines that have communicated with Microsoft Defender ATP cloud on the last 30 days.
 
         Returns:
             dict. Machine's info
         """
-        cmd_url = '/machines'
+        cmd_url = f'/machines?$top={int(page_size)}'
```
Comment (Itay4) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:231]: better to pass this in the params as done below
also the `int` casting is probably not required

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml [hunk 2]
```diff
@@ -195,6 +195,18 @@ script:
       name: hostname
       required: false
       secret: false
+    - name: next_link
+      default: false
+      description: The link for the next page of results, if it exists.
+      isArray: false
+      required: false
+      secret: false
+    - name: page_size
```
Comment (yuvalbenshalom) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.yml:204]: add default value to the argument in the yml

**Review by yuvalbenshalom — COMMENTED** [review]
see my comment

**Comment by bakatzir** [comment]
reference: https://github.com/demisto/etc/issues/34530

**Pull request closed without merging**